RecoVertex/V0Producer/src/V0Fitter.cc

```diff
@@ -116,12 +117,8 @@ void V0Fitter::fitAll(const edm::Event& iEvent, const edm::EventSetup& iSetup,
    // fill vectors of TransientTracks and TrackRefs after applying preselection cuts
    for (reco::TrackCollection::const_iterator iTk = theTrackCollection->begin(); iTk != theTrackCollection->end(); ++iTk) {
       const reco::Track* tmpTrack = &(*iTk);
-      double ipsigXY;
-      if (useVertex_) {
-         ipsigXY = std::abs(tmpTrack->dxy(referencePos)/tmpTrack->dxyError());
-      } else {
-         ipsigXY = std::abs(tmpTrack->dxy(*theBeamSpot)/tmpTrack->dxyError()); 
-      }
+      double ipsigXY = std::abs(tmpTrack->dxy(*theBeamSpot)/tmpTrack->dxyError());
+      if (useVertex_) ipsigXY = std::abs(tmpTrack->dxy(referencePos)/tmpTrack->dxyError());
```

Comment:
On 10/28/15 10:05 AM, Vincenzo Innocente wrote:
> In RecoVertex/V0Producer/src/V0Fitter.cc
> <https://github.com/cms-sw/cmssw/pull/12142#discussion_r43267434>:
> 
>> @@ -116,12 +117,8 @@ void V0Fitter::fitAll(const edm::Event& iEvent, const edm::EventSetup& iSetup,
>>     // fill vectors of TransientTracks and TrackRefs after applying preselection cuts
>>     for (reco::TrackCollection::const_iterator iTk = theTrackCollection->begin(); iTk != theTrackCollection->end(); ++iTk) {
>>        const reco::Track* tmpTrack = &(*iTk);
>> -      double ipsigXY;
>> -      if (useVertex_) {
>> -         ipsigXY = std::abs(tmpTrack->dxy(referencePos)/tmpTrack->dxyError());
>> -      } else {
>> -         ipsigXY = std::abs(tmpTrack->dxy(*theBeamSpot)/tmpTrack->dxyError()); 
>> -      }
>> +      double ipsigXY = std::abs(tmpTrack->dxy(*theBeamSpot)/tmpTrack->dxyError());
>> +      if (useVertex_) ipsigXY = std::abs(tmpTrack->dxy(referencePos)/tmpTrack->dxyError());
> 
> why computing twice ipsingXY in case of useVertex_?
> was not the previous code more efficient?

if/else is probably going to be more CPU efficient
(unclear though how the compiler puts the conditional computations in
optimization)


> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/12142/files#r43267434>.
>